tests/components/airzone/test_coordinator.py

```diff
@@ -64,3 +64,58 @@ async def test_coordinator_client_connector_error(hass: HomeAssistant) -> None:
 
         state = hass.states.get("sensor.despacho_temperature")
         assert state.state == STATE_UNAVAILABLE
+
+
+async def test_coordinator_new_devices(hass: HomeAssistant) -> None:
+    """Test new devices on coordinator update."""
+
+    config_entry = MockConfigEntry(
+        data=CONFIG,
+        domain=DOMAIN,
+        unique_id="airzone_unique_id",
+    )
+    config_entry.add_to_hass(hass)
+
+    with (
+        patch(
+            "homeassistant.components.airzone.AirzoneLocalApi.get_dhw",
+            side_effect=HotWaterNotAvailable,
+        ),
+        patch(
+            "homeassistant.components.airzone.AirzoneLocalApi.get_hvac",
+            return_value=HVAC_MOCK_NEW_ZONES,
+        ) as mock_hvac,
+        patch(
+            "homeassistant.components.airzone.AirzoneLocalApi.get_hvac_systems",
+            side_effect=SystemOutOfRange,
+        ),
+        patch(
+            "homeassistant.components.airzone.AirzoneLocalApi.get_version",
+            return_value=HVAC_VERSION_MOCK,
+        ),
+        patch(
+            "homeassistant.components.airzone.AirzoneLocalApi.get_webserver",
+            side_effect=InvalidMethod,
+        ),
+    ):
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+        mock_hvac.assert_called_once()
+        mock_hvac.reset_mock()
+
+        state = hass.states.get("sensor.salon_temperature")
+        assert state.state == "19.6"
+
+        state = hass.states.get("sensor.dorm_ppal_temperature")
+        assert state is None
+
+        mock_hvac.return_value = HVAC_MOCK
+        async_fire_time_changed(hass, utcnow() + SCAN_INTERVAL)
```

Comment:
Please use the freezer as well

Reply:
Done in https://github.com/home-assistant/core/pull/121891/commits/4b66f3589be4cc44ef5298cb7c07c7f84c6a3b37?w=1